servers/file_system/fat/detect_fat.c

```diff
@@ -98,15 +118,36 @@ bool detect_fat(fat_info_type *fat_info)
         message_parameter.data = &ipc_block_read;
         message_parameter.length = sizeof(ipc_block_read_type);
 
-        ipc_block_read.start_block_number = 1;
+        ipc_block_read.start_block_number = bios_parameter_block->reserved_sectors;
```

Comment:
This is the essence of the change. The other parts is more "added missing error handling", whitespace fixes etc.